builtin/providers/aws/resource_aws_cloudfront_distribution.go

```diff
@@ -530,6 +540,19 @@ func resourceAwsCloudFrontDistributionRead(d *schema.ResourceData, meta interfac
 	d.Set("last_modified_time", aws.String(resp.Distribution.LastModifiedTime.String()))
 	d.Set("in_progress_validation_batches", resp.Distribution.InProgressInvalidationBatches)
 	d.Set("etag", resp.ETag)
+	d.Set("arn", resp.Distribution.ARN)
+
+	cloudFrontArn := resp.Distribution.ARN
+	tagResp, tagErr := conn.ListTagsForResource(&cloudfront.ListTagsForResourceInput{
+		Resource: cloudFrontArn,
+	})
+
+	if tagErr != nil {
+		log.Printf("[DEBUG] Error retrieving tags for ARN: %s", cloudFrontArn)
+	}
+
+	d.Set("tags", tagsToMapCloudFront(tagResp.Tags))
```

Comment:
We can guard against it no worries at all - I was just going with the API description:

```
// The returned result of the corresponding request.
type ListTagsForResourceOutput struct {
	_ struct{} `type:"structure" payload:"Tags"`

	// A complex type that contains zero or more Tag elements.
	Tags *Tags `type:"structure" required:"true"`
}
```

Have pushed the guard :)